### ISSUE TITLE
replace deprecated middleware

### ISSUE THREAD

**Issue by n2ygk** [jazzband/django-oauth-toolkit#799]
Refactor to use "new-style" Django Middleware. Without doing so, this code will break in Django 4.0.  See deprecation warnings like this:

```
tests/test_auth_backends.py::TestOAuth2Middleware::test_middleware_response
  .../django-oauth-toolkit/tests/test_auth_backends.py:122: RemovedInDjango40Warning: Passing None for the middleware get_response argument is deprecated.
    m = OAuth2TokenMiddleware()
```

[This blog post](https://django.cowhite.com/blog/django-middleware-the-new-version-of-middleware-and-how-to-use-it/) describes the change and even uses this project as an example!

**Comment by auvipy** [comment]
ok I will

**Comment by smithdc1** [comment]
Fixed by #862 ?

**Comment by MattBlack85** [comment]
fixed by #862
